core/shared/src/main/scala/org/http4s/MediaType.scala

```diff
@@ -313,7 +313,7 @@ object MediaType extends MimeDB {
   private[http4s] def getMediaType(_mainType: String, _subType: String): MediaType = {
     val mainType = _mainType.toLowerCase
     val subType = _subType.toLowerCase
-    if (Platform.isJvm || Platform.isNative)
+    if (Platform.isJvm)
```

Comment:
Thanks! Probably I will just agree to this in the end 😁 but there is a slight behavior change for recognizing compressible types. There are a couple other strategies we could consider:

1. avoiding linking MimeDB if we are in debug mode (although this could be highly confusing)
2. avoiding linking MimeDB if we are in release-size mode ... but even in ordinary speed-optimized release mode it is annoying and unwanted

Reply:
I personally prefer to avoid linking MimeDB every time.
You get:
- consistency, since all builds behave the same way and you don't have random bugs in production that you don't see in development
- faster builds both in development (I don't want my CIs to take 20 minutes to link an app, I'd rather not use Scala Native at all)